src/BuildingRegistry/Building/Building_BuildingUnit.cs

```diff
@@ -53,32 +54,18 @@ public void RealizeBuildingUnit(BuildingUnitPersistentLocalId buildingUnitPersis
                 BuildingStatus.Retired
             });
 
-            var buildingUnit = BuildingUnits.FirstOrDefault(x => x.BuildingUnitPersistentLocalId == buildingUnitPersistentLocalId);
-
-            if (buildingUnit is null)
-            {
-                throw new BuildingUnitIsNotFoundException(
-                    BuildingPersistentLocalId,
-                    buildingUnitPersistentLocalId);
-            }
-
-            buildingUnit.Realize();
+            _buildingUnits
+                .GetNotRemovedByPersistentLocalId(buildingUnitPersistentLocalId)
```

Comment:
@ArneD doubting if we want to keep or delete the GuardRemoved() on BuildingUnit methods (Realize, NotRealize ...).

Reply:
I'm conflicted. On the one hand you have an aggregate witch is the entry point of your domain and thus the checks are done once entering the domain.
On the other hand if you code in the domain and call `.Realize` I would expect that the `Entity` also protects itself from invalid state / actions.

Reply:
Was having the same thing in mind. Let's for now keep it like that.